Update docs with TLS for migrations. (#4189)
pulumi/docs: content/docs/guides/self-hosted/api.md

#### Database Connections

##### API Service

The service is configurable to enable connections to the backend SQL database over TLS. The following environment variables are _all_ required to connect to the database using TLS. If these variables are set the service will establish connections to the database using TLS, otherwise the service will default to connecting without TLS. The same ports will be used for communication to the database regardless of whether TLS is configured or not.

| Variable Name            | Description                                                                                                   |
|--------------------------|---------------------------------------------------------------------------------------------------------------|
| DATABASE_CA_CERTIFICATE  | The CA certificate used to establish TLS connections with the database. This certificate must be PEM encoded. This must be set to the value of the certificate itself and _not_ a filepath to the location of the certificate file. |
| DATABASE_MIN_TLS_VERSION | The minimum TLS version to use for database connections (must be in \<major>.\<minor> format, e.g. `1.2`).    |

##### Migrations

> The database migrations are configurable to enable connections to the database over TLS. If you are intending to connect to the database over TLS to run the migrations the following environment variable must be set in order to enable this. If this variable is set connections to the database will be establised over TLS, otherwise connections will default to not using TLS.

Reviewer: ```suggestion
The database migrations container is configurable to enable connections to the database over TLS. To use TLS, the following environment variable must be set. The default is to not use TLS.
```

Author: I like it. I think its a lot more concise and reads easier. Thank you.
